tracing-error/src/backtrace.rs

```diff
@@ -120,6 +120,47 @@ impl SpanTrace {
             }
         });
     }
+
+    /// Returns the status of this SpanTrace, indicating whether this backtrace request was unsupported, empty, or a span trace was actually captured.
+    pub fn status(&self) -> SpanTraceStatus {
+        SpanTraceStatus(if self.span.is_none() {
+            SpanTraceStatusInner::Empty
+        } else {
+            let mut status = None;
+            self.span.with_subscriber(|(_, s)| {
+                if s.downcast_ref::<WithContext>().is_some() {
+                    status = Some(SpanTraceStatusInner::Captured);
+                }
+            });
+
+            status.unwrap_or(SpanTraceStatusInner::Unsupported)
+        })
```

Comment:
nit/style: personally, i am really not a fan of wrapping such a large expression in a struct constructor. i would probably do
```suggestion
        let inner = if self.span.is_none() {
            SpanTraceStatusInner::Empty
        } else {
            let mut status = None;
            self.span.with_subscriber(|(_, s)| {
                if s.downcast_ref::<WithContext>().is_some() {
                    status = Some(SpanTraceStatusInner::Captured);
                }
            });

            status.unwrap_or(SpanTraceStatusInner::Unsupported)
        };
        SpanTraceStatus(inner)
```

but, this is not a blocker.